src/FeeRecipientDonationModule.sol

```diff
@@ -0,0 +1,423 @@
+    ////////////////////////////////////////////////////////////////////////////
+    // STORAGE
+    ////////////////////////////////////////////////////////////////////////////
+    address public guardian;
+    address public keeper;
+
+    uint256 public lastProcessingTimestamp;
+    uint256 public annualizedYieldBPS;
+    uint256 public minOutBPS;
+    bytes public swapPath;
+
+    ////////////////////////////////////////////////////////////////////////////
+    // ERRORS
+    ////////////////////////////////////////////////////////////////////////////
+
+    error NotGovernance(address caller);
+    error NotGovernanceOrGuardian(address caller);
+    error NotKeeper(address caller);
+
+    error TooSoon(uint256 lastProcessing, uint256 timestamp);
+
+    error ZeroIntervalPeriod();
+    error ZeroAddress();
+    error ModuleMisconfigured();
+
+    ////////////////////////////////////////////////////////////////////////////
+    // EVENTS
+    ////////////////////////////////////////////////////////////////////////////
+
+    event GuardianUpdated(address indexed oldGuardian, address indexed newGuardian);
+    event KeeperUpdated(address indexed oldKeeper, address indexed newKeeper);
+    event SwapPathUpdated(bytes oldPath, bytes newPath);
+    event AnnualizedYieldUpdated(uint256 oldYield, uint256 newYield);
+    event MinOutUpdated(uint256 oldMinOut, uint256 newMinOut);
+    event PerformedUpkeep(
+        uint256 collSharesToClaim, 
+        uint256 ebtcAmountRequired, 
+        uint256 stEthClaimed, 
+        uint256 wstEthAmount, 
+        uint256 ebtcReceived
+    );
+
+    ////////////////////////////////////////////////////////////////////////////
+    // MODIFIERS
+    ////////////////////////////////////////////////////////////////////////////
+
+    /// @notice Checks whether a call is from governance
+    modifier onlyGovernance() {
+        if (msg.sender != GOVERNANCE) revert NotGovernance(msg.sender);
+        _;
+    }
+
+    /// @notice Checks whether a call is from governance or guardian
+    modifier onlyGovernanceOrGuardian() {
+        if (msg.sender != GOVERNANCE && msg.sender != guardian) revert NotGovernanceOrGuardian(msg.sender);
+        _;
+    }
+
+    /// @notice Checks whether a call is from the keeper.
+    modifier onlyKeeper() {
+        if (msg.sender != keeper) revert NotKeeper(msg.sender);
+        _;
+    }
+
+    /// @param _guardian Address allowed to pause contract
+    constructor(
+        address _guardian, 
+        uint256 _annualizedYieldBPS,
+        uint256 _minOutBPS,
+        bytes memory _swapPath
+    ) {
+        if (_guardian == address(0)) revert ZeroAddress();
+        require(_isValidSwapPath(_swapPath), "bad swapPath");
+
+        guardian = _guardian;
+        annualizedYieldBPS = _annualizedYieldBPS;
+        swapPath = _swapPath;
+
+        minOutBPS = _minOutBPS;
+        require(minOutBPS <= BPS && minOutBPS >= MIN_BPS_LOWER_BOUND);
+
+        // keeper will be set later
+    }
```

Comment:
# QA:
Should we make the module wait a week from the deployment?

This is not a big deal either way

Reply:
the keeper won't be configured until at least a week from now, we should be fine